controllers/machinedeletionremediation_controller.go

```diff
@@ -149,6 +149,36 @@ func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, re
 
 	log.Info("node-associated machine found", "node", remediation.Name, "machine", machine.GetName())
 
+	// Detect if Node name is expected to change after Machine deletion given
+	// the Platform type. In case of BareMetal platform, the name is NOT
+	// expected to change, for other providers, instead, the name is expected
+	// to change.
+	// The providerID contains the platform type as prefix (e.g. baremetal:///...)
+	var providerID string
+	if providerID, err = getMachineProviderID(machine); err != nil {
+		log.Error(err, "could not get providerID")
+		return ctrl.Result{}, err
+	}
+	if providerID == "" {
+		log.Info("machine does not have ProviderID")
+		return ctrl.Result{}, nil
```

Comment:
not sure if this can happen ,but I think we should go on with remediation, not? And set condition to Unknown?